server/service/src/sync/central_data_synchroniser.rs

```diff
@@ -60,7 +62,9 @@ impl CentralDataSynchroniser {
 
             logger.progress(
                 SyncStepProgress::PullCentral,
-                max_cursor - last_cursor_in_batch,
+                // During integration tests got attempt to 'substract with overflow'
+                // There is a chance that max_cursor is lower the last cursor in batch
```

Comment:
Interested to understand how this is possible...

Reply:
I think this if fine though, better to be safe rather than crash...

Reply:
I didn't want to dwell on this for too long (and look through 4d code), but I would suspect max_cursor would be queried before the batch is compiled, and while batch is compiled more central records are added